cpp/include/raft/random/detail/rng_impl.cuh

```diff
@@ -294,7 +294,11 @@ DI void custom_next(
   GenType& gen, OutType* val, GumbelDistParams<OutType> params, LenType idx = 0, LenType stride = 0)
 {
   OutType res = 0;
-  gen.next(res);
+
+  do {
+    gen.next(res);
+  } while (res == OutType(1.0));
```

Comment:
I'm surprised we're not checking for `0.0` here whereas in Laplace, we are checking for that case.
If `0.0` is a possible output of the RNG, then clearly it will lead to `Inf` here in the first `log`, right?

Reply:
I ran the code several times with randomized seeds and Gumbel never failed for `0.0` value. I think I'll put the check anyway.

Reply:
I mean the question really is whether `0.0` is a possible output of the RNG.
If it is, then for a strictly positive `beta`, the final value should be `-Inf`.
If `0.0` is not a possible output of the RNG then we shouldn't do the redundant check, but we shouldn't do it in Laplace either IMO.

Reply:
Otherwise, you can simply test with offsets going from `0 .. 2^25` which should guarantee that you hit each output number once (unless the RNG is biased), and see what you get in each case.
I think it's important that we do this at least once with both philox and PCG just to check which output range we are actually getting (regardless of what we think we should be getting).

Reply:
As per design `0,0` should be possible with PCG at least. I'll do the exhaustive search as you suggested and confirm the behavior.